keras/src/ops/numpy.py

```diff
@@ -7762,3 +7762,120 @@ def histogram(x, bins=10, range=None):
+
+        if not isinstance(indices_or_sections, int):
+            raise TypeError(
+                "Argument `indices_or_sections` must be of type `int`. "
+                f"Received: {indices_or_sections}"
+            )
+        if indices_or_sections <= 0:
+            raise ValueError(
+                "Argument `indices_or_sections` must be a positive integer. "
+                f"Received: {indices_or_sections}"
+            )
+        if not isinstance(axis, int):
+            raise TypeError(
+                f"Argument `axis` must be of type `int`. Received: {axis}"
+            )
+
+        self.indices_or_sections = indices_or_sections
+        self.axis = axis
+
+    def call(self, x):
+        # Call the backend's array_split implementation directly.
+        # It handles the logic for uneven splits.
+        return backend.numpy.array_split(
+            x,
+            indices_or_sections=self.indices_or_sections,
+            axis=self.axis,
+        )
+
+    def compute_output_spec(self, x):
+        num_splits = self.indices_or_sections
+
+        # Normalize axis
+        axis = self.axis
+        if axis < 0:
+            axis += len(x.shape)
+
+        total_size = x.shape[axis]
+
+        if total_size is None:
+            # Dynamic shape: We know the number of splits, but not their sizes.
+            output_specs = []
+            base_shape = list(x.shape)
+            base_shape[axis] = None  # Size of this axis is unknown
+            for _ in range(num_splits):
+                output_specs.append(
+                    KerasTensor(shape=tuple(base_shape), dtype=x.dtype)
+                )
+            return tuple(output_specs)
+
+        # Static shape: We can compute the exact size of each split.
+        split_size = total_size // num_splits
+        remainder = total_size % num_splits
+
+        output_specs = []
+        base_shape = list(x.shape)
+        for i in range(num_splits):
+            size = split_size + (1 if i < remainder else 0)
+            shape = base_shape.copy()
+            shape[axis] = size
+            output_specs.append(KerasTensor(shape=tuple(shape), dtype=x.dtype))
+
+        return tuple(output_specs)
+
+
+@keras_export(["keras.ops.array_split", "keras.ops.numpy.array_split"])
+def array_split(x, indices_or_sections, axis=0):
+    """Splits an array into multiple sub-arrays (unevenly).
+
+    This is similar to `keras.ops.split`, but it allows for
+    unequal splits. `indices_or_sections` must be an integer
+    that indicates the total number of sub-arrays to create.
+    If the tensor cannot be divided evenly, the first `remainder`
+    splits will have size `quotient + 1`, and the rest will
+    have size `quotient`.
+
+    Args:
+        x: Input tensor.
+        indices_or_sections: An integer indicating the number of
+            sub-arrays to create.
+        axis: The axis along which to split. Defaults to 0.
+
+    Returns:
+        A tuple of sub-tensors.
```

Comment:
It should actually be a list of tensors per NumPy. So change the documentation and implementation as needed.